api/src/test/java/jakarta/data/displaynamegeneration/ReplaceCamelCaseAndUnderscoreAndNumber.java

```diff
@@ -0,0 +1,169 @@
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package jakarta.data.displaynamegeneration;
+
+import org.junit.jupiter.api.DisplayNameGenerator;
+
+import java.lang.reflect.Method;
+
+/**
+ * <p>A class extending {@linkplain DisplayNameGenerator.Standard }.</p>
+ *
+ * <p>This extension handles method names with CamelCase, underscore and numbers.</p>
+ *
+ * <p>The aim is to simplify unit test display names. Instead of using this method annotation {@linkplain org.junit.jupiter.api.DisplayName },
+ * we can just use this class annotation {@linkplain org.junit.jupiter.api.DisplayNameGeneration } and use that method annotation if needed.
+ * </p>
+ *
+ * <p>This generator follows 3 rules:</p>
+ *
+ * <ul>
+ *     <li>Each uppercase letter is turned into its lowercase value prepended by space.</li>
+ *     <li>Each underscore is turned into space. Words bounded by underscores or just starting with underscore are not transformed. Usually these words represent classes, variables....</li>
+ *     <li>Each number is prepended by space.</li>
+ * </ul>
+ *
+ * Usage example:
+ *
+ * <pre>
+ *
+ *     {@code
+ *
+ * @DisplayNameGeneration(ReplaceCamelCaseAndUnderscoreAndNumber.class)
+ * class ExampleTest {
+ *     @Test
+ *     //Equivalent of @DisplayName("Should return error when maxResults is negative")
+ *     void shouldReturnErrorWhen_maxResults_IsNegative() {
+ *       ...
+ *     }
+ *     @Test
+ *     //Equivalent of @DisplayName("Should create limit with range")
+ *     void shouldCreateLimitWithRange() {
+ *       ...
+ *     }
+ *
+ *     @Test
+ *     //Equivalent of @DisplayName("Should return 5 errors")
+ *     void shouldReturn5Errors() {
+ *       ...
+ *     }
+ *
+ *     @Test
+ *     //Equivalent of @DisplayName("Should return the value of maxResults")
+ *     void shouldReturnTheValueOf_maxResults() {
+ *       ...
+ *     }
+ *
+ *     @Test
+ *     //Equivalent of @DisplayName("Should return the number of errors as numberOfErrors inferior or equal to 15")
+ *     void shouldReturnTheNumberOfErrorsAs_numberOfErrors_InferiorOrEqualTo15() {
+ *       ...
+ *     }
+ * }
+ *
+ *     }
+ * </pre>
+ *
+ * */
+public class ReplaceCamelCaseAndUnderscoreAndNumber extends DisplayNameGenerator.Standard {
+
+    //TODO: Use after making String replaceCamelCaseAndUnderscoreAndNumber(String input) private
+    static final DisplayNameGenerator INSTANCE = new ReplaceCamelCaseAndUnderscoreAndNumber();
+
+    public ReplaceCamelCaseAndUnderscoreAndNumber() {
+        super();
+    }
```

Comment:
I don't think any of this is necessary if we are just extending a class to replace a method.

Reply:
ok, will update